iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/processor/downsampling/DownSamplingProcessor.java

```diff
@@ -138,6 +202,37 @@ protected abstract void processRow(
       String deviceSuffix,
       AtomicReference<Exception> exception);
 
+  /**
+   * Determine the arrival time interval and event time interval.
+   *
+   * @return true to indicate that the process does not need to be continued and the data can be
+   *     updated directly. False means that the event is discarded directly. Null means that the
+   *     downsampling algorithm can continue to be executed.
+   */
+  protected Boolean filterArrivalTimeAndEventTime(
+      final DownSamplingFilter filter, final long arrivalTime, final long eventTime) {
+    final long arrivalTimeInterval = Math.abs(arrivalTime - filter.getLastPointArrivalTime());
```

Comment:
May need further discussion about the precision of the arrival time... Currently the doc declare it as the same as the IoTDB's precision..